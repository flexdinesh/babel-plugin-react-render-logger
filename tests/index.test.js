import { transformSync } from '@babel/core';
import reactRenderLogger from '../src/index';

const getTestConfig = (opts = {}) => ({
  plugins: ['@babel/plugin-transform-react-jsx', [reactRenderLogger, { disableConsoleStyles: true, ...opts }]]
});

describe('Add log statement to React Components', () => {
  test('ArrowFunctionExpression with BlockStatement', () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());
    expect(code).toMatchSnapshot();
  });

  test('ArrowFunctionExpression with implicit return', () => {
    const codeLiteral = `
      const HelloWorld = (props) => <span>Hello World</span>;
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());

    expect(code).toMatchSnapshot();
  });

  test('FunctionDelaration with BlockStatement', () => {
    const codeLiteral = `
      function HelloWorld() {
        return <span>Hello World</span>;
      }
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());

    expect(code).toMatchSnapshot();
  });

  test('FunctionExpression with BlockStatement', () => {
    const codeLiteral = `
      const HelloWorld = function() {
        return <span>Hello World</span>;
      }
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());

    expect(code).toMatchSnapshot();
  });

  test('Class Component with render method', () => {
    const codeLiteral = `
      class HelloWorld extends React.Component {
        render() {
          return <span>Hello World</span>
        }
      }
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());

    expect(code).toMatchSnapshot();
  });
});

describe('Add log statement based on config', () => {
  test('name matches full component name - should include logger', () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const opts = { name: 'HelloWorld' };
    const { code } = transformSync(codeLiteral, getTestConfig(opts));
    expect(code).toMatchSnapshot();
  });

  test('name matches part component name - should include logger', () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const opts = { name: 'Hello' };
    const { code } = transformSync(codeLiteral, getTestConfig(opts));
    expect(code).toMatchSnapshot();
  });

  test("name doesn't match component name - should not include logger", () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const opts = { name: 'Oohlaalaa' };
    const { code } = transformSync(codeLiteral, getTestConfig(opts));
    expect(code).toMatchSnapshot();
  });

  test('name matches multiple strings - should include logger', () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const opts = { name: 'Hello|Kitty' };
    const { code } = transformSync(codeLiteral, getTestConfig(opts));
    expect(code).toMatchSnapshot();
  });

  test('name is not passed - should include logger', () => {
    const codeLiteral = `
      const HelloWorld = () => {
        return <span>Hello World</span>;
      };
    `;
    const { code } = transformSync(codeLiteral, getTestConfig());
    expect(code).toMatchSnapshot();
  });
});
