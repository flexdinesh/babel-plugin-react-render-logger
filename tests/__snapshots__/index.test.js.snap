// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add log statement based on config name doesn't match component name - should not include logger 1`] = `
"const HelloWorld = () => {
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement based on config name is not passed - should include logger 1`] = `
"const HelloWorld = () => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement based on config name matches full component name - should include logger 1`] = `
"const HelloWorld = () => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement based on config name matches multiple strings - should include logger 1`] = `
"const HelloWorld = () => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement based on config name matches part component name - should include logger 1`] = `
"const HelloWorld = () => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement to React Components ArrowFunctionExpression with BlockStatement 1`] = `
"const HelloWorld = () => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement to React Components ArrowFunctionExpression with implicit return 1`] = `
"const HelloWorld = props => {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;

exports[`Add log statement to React Components Class Component with render method 1`] = `
"class HelloWorld extends React.Component {
  render() {
    console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
    return React.createElement(\\"span\\", null, \\"Hello World\\");
  }

}"
`;

exports[`Add log statement to React Components FunctionDelaration with BlockStatement 1`] = `
"function HelloWorld() {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
}"
`;

exports[`Add log statement to React Components FunctionExpression with BlockStatement 1`] = `
"const HelloWorld = function () {
  console.log(\\">> RenderLog >>\\", \\"HelloWorld\\");
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;
